docker/service.go

```diff
@@ -300,14 +278,28 @@ func (s *Service) up(ctx context.Context, imageName string, create bool, options
 		containers = []*Container{c}
 	}
 
-	return s.eachContainer(ctx, containers, func(c *Container) error {
-		if create {
-			if err := s.recreateIfNeeded(ctx, imageName, c, options.NoRecreate, options.ForceRecreate); err != nil {
+	hasBeenRecreate := false
+
+	return s.watchContainers(ctx, containers, func(c *Container) error {
+		if !hasBeenRecreate {
+			hasBeenRecreate = true
+			if create {
+				if err := s.recreateIfNeeded(ctx, imageName, c, options.NoRecreate, options.ForceRecreate); err != nil {
+					return err
+				}
+			}
+			if err := c.Up(ctx, imageName); err != nil {
 				return err
 			}
 		}
-
-		return c.Up(ctx, imageName)
+		if options.Log {
+			return c.Log(ctx, true)
+		}
+		return nil
+	}, func(c *Container) error {
+		// FIXME(vdemeester) add status code & co
```

Comment:
This `FIXME` will be handled with the `Container` struct refactoring :stuck_out_tongue_closed_eyes: